pkg/pool-weighted/contracts/InvariantGrowthProtocolFees.sol

```diff
@@ -26,6 +28,12 @@ abstract contract InvariantGrowthProtocolFees is BaseWeightedPool {
     // This mechanism requires keeping track of the invariant after the last join or exit.
     uint256 private _lastPostJoinExitInvariant;
 
+    constructor(IProtocolFeePercentagesProvider protocolFeeProvider)
```

Comment:
Because the behavior of WeightedPool and ManagedPool will diverge (ManagedPool needs to have the fixed fee option) - and to conform to the "initialize at the most derived level" convention (e.g., from Stable) - I didn't have a constructor here. InvariantGrowthProtocolFees does derive from ProtocolFeeCache, but I initialize it in WeightedPool

Reply:
While I agree with @EndymionJkb in this regard, we don't need to worry right now about what the final Managed Pool will look like, and can focus instead on finishing the current iteration of Weighted.